feat: improve HTTP API logging, honor no-startup-message

jinnatar opened the pull request:
They provide little useful information outside debugging and anything at
Info level is broadcast over shoutrrr.

simonaronsson reviewed (changes requested):
I agree with @piksel on this.

jinnatar committed:
Move HTTP API startup logging to be affected by --no-startup-message

While the log message is useful in seeing the HTTP API is enabled and
functioning, in production installations it should be silenced to
prevent notifications going out on every startup with the API enabled.

jinnatar committed:
Show startup messages also when running in non-polling API mode

To prevent these getting sent as notifications, use --no-startup-message

jinnatar committed:
Update startup logging to be more accurate for a disabled scheduler.

Previously the info was logged only if a single run was requested but
now also the default API mode has a zero length schedule.

piksel commented on cmd/root.go:
```diff
@@ -159,6 +159,7 @@ func Run(c *cobra.Command, names []string) {
 	enableMetricsAPI, _ := c.PersistentFlags().GetBool("http-api-metrics")
 	unblockHTTPAPI, _ := c.PersistentFlags().GetBool("http-api-periodic-polls")
 	apiToken, _ := c.PersistentFlags().GetString("http-api-token")
+	enableAPIPeriodicPolls, _ := c.PersistentFlags().GetBool("http-api-periodic-polls")
```
Why make another copy of the flag?

jinnatar replied:
Because I was being blind and didn't notice `unblockHTTPAPI ` was already there. Will fix.

piksel commented on pkg/api/api.go:
```diff
@@ -71,6 +70,6 @@ func (api *API) Start(block bool) error {
 
 func runHTTPServer() {
-	log.Info("Serving HTTP")
+	log.Debug("Serving HTTP")
```
```suggestion
```
I don't think this serves any purpose now

piksel commented on cmd/root.go:
```diff
@@ -287,7 +294,12 @@ func writeStartupMessage(c *cobra.Command, sched time.Time, filtering string) {
 		startupLog.Info("Note that the first check will be performed in " + until)
 	} else {
-		startupLog.Info("Running a one time update.")
+		startupLog.Info("Periodic runs are not enabled.")
+	}
```
This is a bit weird when just starting with `--run-once`, as it will be followed by said update.

```suggestion
		if runOnce, _ := c.PersistentFlags().GetBool("run-once"); runOnce {
			startupLog.Info("Running a one time update.")
        	} else {
			startupLog.Info("Periodic runs are not enabled.")
		}
	}
```

It should probably be `else if` etc. but that's not supported by GitHub suggestions...

jinnatar replied:
It made sense to me, but you could be right that not to anyone else so better to be more helpful. Added the else if version.

piksel reviewed (changes requested):
Looks good, except for a few smaller things.

The API parts are messy and should be redone (ref #939). But this should be sufficient for now. Just take a look at the comments.

jinnatar committed:
Drop useless debug logging when starting HTTP endpoint

Co-authored-by: nils måsén <nils@piksel.se>

jinnatar committed:
Make --run-once startup message more helpful, avoid code duplication
